packages/owl-bot-post-processor/src/core.ts

```diff
@@ -0,0 +1,299 @@
+  // in practice this should be rare:
+  const {data: commits} = await octokit.pulls.listCommits({
+    owner,
+    repo,
+    pull_number: pr,
+    per_page: 100,
+  });
+  const headCommit = commits[commits.length - 1];
+  if (!headCommit) return;
+  else return headCommit as Commit;
+}
+
+export async function createCheck(
+  args: CheckArgs,
+  octokit?: OctokitType,
+  logger = console
+) {
+  if (!octokit) {
+    octokit = await core.getAuthenticatedOctokit({
+      'pem-path': args['pem-path'],
+      'app-id': args['app-id'],
+      installation: args.installation,
+    });
+  }
+  const [owner, repo] = args.repo.split('/');
+  const headCommit = await getHeadCommit(owner, repo, Number(args.pr), octokit);
+  if (!headCommit) {
+    logger.warn(`no commit found for PR ${args.pr}`);
+    return;
+  }
+  await octokit.checks.create({
+    owner,
+    repo,
+    name: 'OwlBot Post Processor',
+    summary: args.summary,
+    head_sha: headCommit.sha as string,
+    conclusion: args.conclusion,
+    output: {
+      title: args.title,
+      summary: args.summary,
+      text: args.text,
+    },
+  });
+}
+
+export async function getGitHubShortLivedAccessToken(
+  pemPath: string,
+  appId: number,
+  installation: string
+): Promise<Token> {
+  const privateKey = await readFileAsync(pemPath);
+  const payload = {
+    // issued at time
+    // Note: upstream API seems to fail if decimals are included
+    // in unixtime, this is why parseInt is run:
+    iat: parseInt('' + Date.now() / 1000),
+    // JWT expiration time (10 minute maximum)
+    exp: parseInt('' + Date.now() / 1000 + 10 * 60),
+    // GitHub App's identifier
+    iss: appId,
+  };
+  const jwt = sign(payload, privateKey, {algorithm: 'RS256'});
+  const resp = await request<Token>({
+    url: getAccessTokenURL(installation),
+    method: 'POST',
+    headers: {
+      Authorization: `Bearer ${jwt}`,
+      Accept: 'application/vnd.github.v3+json',
+    },
+  });
+  if (resp.status !== 201) {
+    throw Error(`unexpected response http = ${resp.status}`);
+  } else {
+    return resp.data;
+  }
+}
+
+export function getAccessTokenURL(installation: string) {
+  return `https://api.github.com/app/installations/${installation}/access_tokens`;
+}
+
+let cachedOctokit: OctokitType;
+export async function getAuthenticatedOctokit(
+  auth: string | AuthArgs,
+  cache = true
+): Promise<OctokitType> {
+  if (cache && cachedOctokit) return cachedOctokit;
+  let tokenString: string;
+  if (auth instanceof Object) {
+    const token = await getGitHubShortLivedAccessToken(
+      auth['pem-path'],
+      auth['app-id'],
+      auth.installation
+    );
+    tokenString = token.token;
+  } else {
+    tokenString = auth;
+  }
+  const octokit = new Octokit({
+    auth: tokenString,
+  });
+  if (cache) cachedOctokit = octokit;
+  return octokit;
+}
+
+function getCloudBuildInstance() {
+  return new CloudBuildClient();
+}
+
+export const core = {
```

Comment:
This makes it easier to mock specific functions in test, since:

* TypeScript doesn't allow you to reference `exports.functionName`.
* It seems like `module.exports` has read only properties these days.